### ISSUE TITLE
Memory leak problem

### ISSUE THREAD

**Issue by psnoonan** [nuxt-community/axios-module#89]
I have a site with `nuxt:1.2.1` and `nuxtjs/axios:4.5.2` that has been live for 5 days with no problems. I upgraded to `nuxtjs/axios:5.0.0-rc.2` yesterday, also adding a timeout to every request, and immediately saw a memory problem.

I thought the timeout might be causing the issue, so I removed that code and memory stabilized around 50%, which you can see at the tail-end of the attached image.

I continued reading on the issue and decided to roll back my axios version to `nuxtjs/axios:4.5.2` and memory immediately went down to 25%.

![memory-leak](https://user-images.githubusercontent.com/7296414/35699200-922c8ede-074c-11e8-81d2-87fccfcb11bf.PNG)

nuxt.config.js without memory issue
```
    build: {
        vendor: [
            'vue-lazyload',
            'vue-scrollto',
            '~/plugins/helpers.js',
        ],
    },
    plugins: [
        '~/plugins/global-components.js',
        '~/plugins/vue-lazyload.js',
        { src: '~/plugins/vue-scrollto', ssr: false },
    ],
    modules: [
        [
            '@nuxtjs/axios',
            {
                baseURL: 'https://api.blank.com/',
                credentials: false,
            },
        ],
        [
            '@nuxtjs/google-analytics',
            {
                id: 'blank',
                autotracking: {
                    untracked: false,
                },
            },
        ],
        [
            '@nuxtjs/google-tag-manager',
            {
                id: 'blank',
            },
        ],
    ],
```

nuxt.config.js with memory issue (only difference is credentials, and axios-timeout plugin)
```
    build: {
        vendor: [
            'vue-lazyload',
            'vue-scrollto',
            '~/plugins/helpers.js',
        ],
    },
    plugins: [
        '~/plugins/axios-timeout.js',
        '~/plugins/global-components.js',
        '~/plugins/vue-lazyload.js',
        { src: '~/plugins/vue-scrollto', ssr: false },
    ],
    modules: [
        [
            '@nuxtjs/axios',
            {
                baseURL: 'https://api.blank.com/',
            },
        ],
        [
            '@nuxtjs/google-analytics',
            {
                id: 'blank',
                autotracking: {
                    untracked: false,
                },
            },
        ],
        [
            '@nuxtjs/google-tag-manager',
            {
                id: 'blank',
            },
        ],
    ],
```

~/plugins/axios-timeout.js
```
export default ({ $axios }) => {
    $axios.onRequest((config) => { config.timeout = 20000; });
};
```

**Comment by pi0** [comment]
Hey. Thanks for detailed information and help. I will try to investigate more. But this is strange that problem is with `credentials: false` because it is already [disabled by default](https://github.com/nuxt-community/axios-module/blob/master/lib/module.js#L38). And also we have the same implementation of credentials interceptor. What if you disable new `progress` option. I think it may be the only reason to introduce a new leak in 5.x.

**Comment by psnoonan** [comment]
@pi0 After your request, I started adding in the things that seemed to be causing the memory issues, one by one and letting them site for a few hours.

I updated the module version to 5.0.0-rc2, and there were no issues.
Then I added the timeout code back, and there were no issues.

I guess the memory problem was only from the listener in the created lifecycle event, and not from nuxtjs/axios.

Sorry!

I will also post a message in https://github.com/nuxt/nuxt.js/issues/1695 in reply to the concerns I posted there.
